File: zd_05.py
from math import *

g = 9.81

def y(T, U, H):
    return H - (g * T**2) / 2.0

def dy(T):
    return -g * T

def newton_method(U, H, T_initial, epsilon=1e-6, max_iter=100):
    T = T_initial
    for i in range(max_iter):
        y_val = y(T, U, H)
        dy_val = dy(T)
        
        if dy_val == 0:
            print("Derivative is zero.")
            return -1
            
        T_new = T - y_val / dy_val
        
        if abs(T_new - T) < epsilon:
            return T_new
            
        T = T_new

        
    print(f"Did not converge in {max_iter} iterations")
    return -1

H = 50
U = 400

T_initial = sqrt(2 * H / g)
epsilon = 1e-6


T = newton_method(U, H, T_initial, epsilon)

if T != -1:
    print(f"Bomb flight time T: {T} seconds")
    
    t = float(input("Enter time t (in seconds): "))
    x = U * t
    
    print(f"Horizontal distance x(t): {x} meters")

File: zd_05.cpp
#include <iostream>
#include <cmath>
#include <iomanip>

using namespace std;

const double g = 9.81;

double y(double T, double U, double H) {
    return H - (g * T * T) / 2.0;
}

double dy(double T) {
    return -g * T;
}


double newton_method(double U, double H, double T_initial, double epsilon = 1e-6, int max_iter = 100) {
    double T = T_initial;
    for (int i = 0; i < max_iter; ++i) {
        double y_val = y(T, U, H);
        double dy_val = dy(T);
        
        if (dy_val == 0) {
            cout << "Derivative is zero." << endl;
            return -1;
        }
        
        double T_new = T - y_val / dy_val;
        
        if (abs(T_new - T) < epsilon) {
            return T_new;
        }
        
        T = T_new;
    }
    
    cout << "Did not converge in " << max_iter << " iterations" << endl;
    return -1;
}

int main() {
    double H = 50;    
    double U = 400;   
    
    double T_initial = sqrt(2 * H / g);
    double epsilon = 1e-6;
    
    double T = newton_method(U, H, T_initial, epsilon);
    
    if (T != -1) {
        cout << fixed << setprecision(6);
        cout << "Bomb flight time T: " << T << " seconds" << endl;
        
        double t;
        cout << "Enter time t (in seconds): ";
        cin >> t;
        
        double x = U * t;
        cout << "Horizontal distance x(t): " << x << " meters" << endl;
    }
    
    return 0;
}
